controllers/users/users.js

```diff
@@ -118,3 +118,41 @@ exports.deleteUser = (req, res) => {
     .then(() => res.json({ msg: messages.USER_DELETED }))
     .catch(err => res.status(500).send(messages.SERVER_ERROR));
 };
+
+exports.updateUserData = async (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array({ onlyFirstError: true }) });
+  }
+
+  const user = await User.findById(req.user.id);
+  if (user){
+    let { firstName, lastName, email, password } = req.body;
+
+    user.firstName = firstName || user.firstName;
+    user.lastName = lastName || user.lastName;
+
+    if (email){
+      email = email.toLowerCase();
+      if (email !== user.email){
+        const emailExist = await User.findOne({email});
+        if (emailExist){
+          return res.status(400).json({ errors: [{ msg: messages.EMAIL_ALREADY_EXISTS }] });
+        }
+        user.email = email;
+        user.activationKey = bcrypt.hashSync(email).replace(/\//g, '');
+        user.emailVerified = false;
+        sendVerificationEmail(user)
+      }
+    }
+    if (password){
+      user.password = bcrypt.hashSync(password)
+    }
+    await user.save();
+    return res.json({msg: messages.USER_UPDATED })
+  }else {
+    return res.status(500).send(messages.SERVER_ERROR)
+  }
+};
```

Comment:
maybe we should use pattern like this:

```
try {
    const user = await User.findById(req.user.id);

    if (!user) {
      return res.status(400).send(messages.CANT_FIND_USER);
    }
    ...
   
  } catch {
    return res.status(500).send(messages.SERVER_ERROR);
  }
```

We should decide if we'll be using async await syntax or 'then catch' and refractor others routes if need for better readability

Reply:
I think that async await will be clearer